### ISSUE TITLE
Driver silently dumps all but first server in DSN

### ISSUE THREAD

**Issue by jeteon** [crate/crate-pdo#42]
The [docs](https://crate.io/docs/reference/pdo/en/0.4.0/usage.html) suggest that you should be able to use the driver with multiple Crate servers by using a DSN of the form:

```
crate:c1.example.com:4200,c2.example.com:4200,c3.example.com:4200
```

However, the driver just discards all but the first server and uses that for all connections, which happens [here](https://github.com/crate/crate-pdo/blob/master/src/Crate/PDO/PDO.php#L152) in the `parseDSN()` function. I guess its a case of the docs getting ahead of the functionality? I was building an application that assumed the driver would handle fail-over for me only to discover it wouldn't.

I think that the documentation should definitely be updated to reflect that the driver can't handle multi-server DSNs. It would be nice if the driver did behave as promised though. I'll try to put some time into it a little later.

**Comment by chaudum** [comment]
hi @jeteon 
thanks for reporting the issue. the functionality should be available as stated in the docs. we'll have a look and let you know when it's fixed.

**Comment by bogensberger** [comment]
hi @jeteon 

i tried to reproduce the issue with the newest version, but in my case the round robin works correctly and failing nodes where disabled properly.

Which version are you using?

**Comment by jeteon** [comment]
I was using version `0.3.1`. The PDO [docs](https://crate.io/docs/reference/pdo/installation.html) currently suggest using this one. I switched to `0.4.0` since this is the latest stable one available. I can confirm that the issue is indeed no longer present.

I would highly recommend updating the docs on the main site to suggest using `0.4.0` rather than the older `0.3.1`.

I realized I was incorrect before about the source of the problem. It was actually the regex used to parse the DSN.
